### PR TITLE
feat(modal): remove jQuery from Modal component

### PR DESCRIPTION
**Pull request by willamesoares**

### CODE REVIEW

**Comment by willamesoares** [comment]
There are three tests still failing in this PR and they are from the `Confirm` component since the `Modal` is a dependency for the `Confirm` component.

Probably, this Pull Request should be merged when we are done refactoring `Confirm` as well.

File: src/js/components/modal.js [hunk 1]
```diff
@@ -23,12 +25,13 @@ const DEFAULTS = {
...
   constructor (element, options) {
-    this.$element = (element instanceof $) ? element : $(element)
-    this.options = $.extend({}, DEFAULTS, (options || {}))
+    this.element = element
+    this.options = { ...DEFAULTS, ...options }
   }
 
   init () {
-    this.$container = $(this.options.container)
+    this.container = document.querySelector(this.options.container)
```
Comment (fabricionaweb) [inline src/js/components/modal.js]: Depois precisamos ver sobre querySelector vs querySelectorAll

File: src/js/components/modal.js [hunk 2]
```diff
@@ -45,16 +48,18 @@ class Modal {
...
 
   destroy () {
-    this.$element.removeData(NAME)
-    this.$modal.remove()
+    this.element.removeAttribute(`data-${NAME}`)
+    this.modal.remove()
   }
 
   bindListeners () {
     if (this.options.triggerClose) {
```
Comment (fabricionaweb) [inline src/js/components/modal.js]: Acho q rola o destructing dessa triggerClose

File: src/js/components/modal.js (hunk 1 above)
Comment (willamesoares) [reply src/js/components/modal.js]: Nesse caso, vou deixar `querySelector` tbm

**Pull request merged**